### PR TITLE
Push the multi-architecture container images into Docker Hub when tagged

### LINKED ISSUE #413: Automate container image push to Docker Hub
Container images should be loaded into Docker Hub according to the semantic version update.
See https://github.com/docker/build-push-action for more information.

### PR DESCRIPTION
**Pull request by t25kim**
Signed-off-by: Taewan Kim <t25.kim@samsung.com>

# Description

Push the multi-architecture container images into Docker Hub automatically when tagged.

Fixes #413

## Type of change

- [x] CI system update

# How Has This Been Tested?

Add a tag to github master.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by t25kim** [comment]
- See the [Docker Hub result](https://hub.docker.com/repository/registry-1.docker.io/t25kim/edge-home-orchestration-go/tags?page=1&ordering=last_updated)
- See the [workflow result](https://github.com/t25kim/edge-home-orchestration-go/runs/4208702348?check_suite_focus=true)

@suresh-lc @NiranjanPatil25 Please set `homeedge1` account information to `lf-edge/edge-home-orchestration-go`'s Secrets(
DOCKERHUB_USERNAME, DOCKERHUB_PASSWORD).

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          go-version: '1.16.6'
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: lfedge/edge-home-orchestration-go
```
Comment (t25kim) [inline .github/workflows/publish.yml:26]: If someone wants to test this from your Github account, you must change the lfedge to your Docker Hub ID. 
```suggestion
          images: <Docker Hub ID>/edge-home-orchestration-go
```

**Comment by MoonkiHong** [comment]
@lf-edge/edge-home-orchestration-go-maintainers This PR turns from WIP to the request for review. Please take a look. 🙏

**Review by MoonkiHong — APPROVED** [review]
After carefully reviewing all the references including DockerHub, GH action result, it looks good to me. 💯

**Review by tdrozdovsky — APPROVED** [review]
LGTM! 

I checked everything perfectly works!

**Comment by tdrozdovsky** [comment]
> @suresh-lc @NiranjanPatil25 Please set `homeedge1` account information to `lf-edge/edge-home-orchestration-go`'s Secrets(
> DOCKERHUB_USERNAME, DOCKERHUB_PASSWORD). [Reference](https://docs.github.com/en/actions/security-guides/encrypted-secrets)

Done!

**Comment by MoonkiHong** [comment]
@NiranjanPatil25 @suresh-lc Any opinion?

**Review by suresh-lc — APPROVED** [review]
Changes are fine, should we mention somewhere for what all architectures images would be made currently

**Pull request merged**